### PR TITLE
VS Code: enable Fireworks tracing for Sourcegraph teammates

### PR DESCRIPTION
**Pull request by valerybugakov**
- ~~TODO: integrate with chat clients~~.
- Enables Fireworks tracing for autocomplete and chat only for Sourcegraph teammates connected to DotCom.
- Closes https://linear.app/sourcegraph/issue/CODY-2556/client-set-x-fireworks-genie-http-header-to-true-for-members-of-the

## Test plan

- Updated unit tests.
- Tested locally with autocomplete and chat connected to the DotCom instance.

### CODE REVIEW

File: agent/src/index.test.ts [hunk 1]
```diff
@@ -848,7 +848,10 @@ describe('Agent', () => {
         })
     })
 
-    describe('RateLimitedAgent', () => {
+    // Skip `pnpm update-agent-recording` fails with:
+    // AssertionError: expected false to be truthy
+    // expect(serverInfo.authStatus?.isLoggedIn).toBeTruthy()
+    describe.skip('RateLimitedAgent', () => {
```
Comment (valerybugakov) [inline agent/src/index.test.ts:854]: Unsure why this is happening for me locally. Disabled for now. 
@olafurpg, let me know if you have ideas on how to fix it.

Comment (olafurpg) [reply agent/src/index.test.ts:854]: Looking at this test case, we are still using @vdavid 's personal account in the tests haha 😅 I'm guessing his account is no longer rate limited, or the token expired. It's OK to disable this test for now.

**Review by olafurpg — APPROVED** [review]
LGTM 👍🏻 This is fantastic, thank you!

File: agent/src/index.test.ts [hunk 2]
```diff
@@ -917,7 +917,9 @@ describe('Agent', () => {
             expect(lastMessage?.text?.trim()).toStrictEqual('Yes')
         }, 20_000)
 
-        it('commands/document (enterprise client)', async () => {
+        // Skip because it consistently fails with:
+        // Error: Test timed out in 20000ms.
+        it.skip('commands/document (enterprise client)', async () => {
```
Comment (olafurpg) [inline agent/src/index.test.ts:922]: There is something wrong going on here, this test should be instant.

Comment (valerybugakov) [reply agent/src/index.test.ts:922]: Issue to re-enable it later: https://linear.app/sourcegraph/issue/CODY-2622/re-enable-commandsdocument-enterprise-client-agent-test

File: agent/src/index.test.ts (hunk 1 above)
Comment (valerybugakov) [reply agent/src/index.test.ts:854]: Issue to re-enable it: https://linear.app/sourcegraph/issue/CODY-2623/re-enable-ratelimitedagent-test.

**Pull request merged**